src/Kunstmaan/AdminListBundle/Service/EntityVersionLockService.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Kunstmaan\AdminListBundle\Service;
+
+use Doctrine\Common\Persistence\ObjectManager;
+use FOS\UserBundle\Model\User;
+use Kunstmaan\AdminListBundle\Entity\LockableEntity;
+use Kunstmaan\AdminListBundle\Entity\EntityVersionLock;
+use Kunstmaan\AdminListBundle\Repository\EntityVersionLockRepository;
+
+class EntityVersionLockService
+{
+    /**
+     * @var ObjectManager
+     */
+    private $objectManager;
+
+    /**
+     * @var integer
+     */
+    private $threshold;
+
+    /**
+     * @var boolean
+     */
+    private $lockEnabled;
+
+    public function __construct(ObjectManager $em, $threshold, $lockEnabled)
+    {
+        $this->setObjectManager($em);
+        $this->setThreshold($threshold);
+        $this->setLockEnabled($lockEnabled);
+    }
+
+    /**
+     * @param LockableEntity $entity
+     * @return bool
+     */
+    public function isEntityBelowThreshold(LockableEntity $entity)
+    {
+        if ($this->lockEnabled) {
+            $now = date(
```

Comment:
$thresholdDate = clone $entity->getUpdated();
$thresholdDate->add(new \DateInterval("PT{$this->threashold}S"));
$now = new \DateTime('now');

return $thresholdDate >= $now;